### PR TITLE
Added copy constructor to ResourceLimitedVector. [6247]

### PR DESCRIPTION
**Pull request by richiware**
This PR fixed an assertion failure. The application fails with next log:

```
[INFO] [showimage]: Subscribing to topic 'image'
showimage: /home/ricardo/workspace/eprosima/ros2/eloquent/src/eProsima/Fast-RTPS/include/fastrtps/rtps/writer/../common/../../utils/collections/ResourceLimitedVector.hpp:341: bool eprosima::fastrtps::ResourceLimitedVector<_Ty, _KeepOrderEnabler, _LimitsConfig, _Alloc, _Collection>::ensure_capacity() [with _Ty = eprosima::fastrtps::rtps::Locator_t; _KeepOrderEnabler = std::integral_constant<bool, false>; _LimitsConfig = eprosima::fastrtps::ResourceLimitedContainerConfig; _Alloc = std::allocator<eprosima::fastrtps::rtps::Locator_t>; _Collection = std::vector<eprosima::fastrtps::rtps::Locator_t>]: La declaración `configuration_.increment > 0' no se cumple.
```

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/285/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/260/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/283/)

File: include/fastrtps/utils/collections/ResourceLimitedVector.hpp [hunk 1]
```diff
@@ -95,6 +95,14 @@ class ResourceLimitedVector
         collection_.reserve(cfg.initial);
     }
 
+    ResourceLimitedVector(const ResourceLimitedVector& other)
```
Comment (MiguelCompany) [inline include/fastrtps/utils/collections/ResourceLimitedVector.hpp]: Parameter on next line

Comment (richiware) [reply include/fastrtps/utils/collections/ResourceLimitedVector.hpp]: done

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/291/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/266/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/289/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/293/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/268/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/291/)

**Comment by richiware** [comment]
Failed test not related.

**Pull request merged**